manim/mobject/types/vectorized_mobject.py

```diff
@@ -2229,31 +2272,106 @@ def construct(self):
     """
 
     def __init__(
-        self, vmobject, num_dashes=15, dashed_ratio=0.5, color=WHITE, **kwargs
+        self,
+        vmobject,
+        num_dashes=15,
+        dashed_ratio=0.5,
+        dash_offset=0,
+        color=WHITE,
+        even_lengths=True,
+        **kwargs,
     ):
 
         self.dashed_ratio = dashed_ratio
         self.num_dashes = num_dashes
         super().__init__(color=color, **kwargs)
         r = self.dashed_ratio
         n = self.num_dashes
-        if num_dashes > 0:
+        if n > 0:
             # Assuming total length is 1
             dash_len = r / n
             if vmobject.is_closed():
                 void_len = (1 - r) / n
             else:
-                void_len = (1 - r) / (n - 1)
+                if n == 1:
+                    void_len = 1 - r
+                else:
+                    void_len = (1 - r) / (n - 1)
 
-            self.add(
-                *(
-                    vmobject.get_subcurve(
-                        i * (dash_len + void_len),
-                        i * (dash_len + void_len) + dash_len,
-                    )
+            period = dash_len + void_len
+            phase_shift = (dash_offset % 1) * period
+
+            # closed shapes can handle overflow at the 0-point
+            if vmobject.is_closed():
+                dash_starts = [((i * period + phase_shift) % 1) for i in range(n)]
+                dash_ends = [
+                    ((i * period + dash_len + phase_shift) % 1) for i in range(n)
+                ]
+            else:
+                dash_starts = [
+                    ((i * period + phase_shift) % (1 + void_len)) for i in range(n)
+                ]
+                dash_ends = [
+                    ((i * period + dash_len + phase_shift) % (1 + void_len))
                     for i in range(n)
+                ]
```

Comment:
I wonder whether this would be slightly less repetitive if this would be something like
```py
period_len = 1
if not vmobject.is_closed():
    period_len += void_len

dash_starts = [... % period_len ...]
dash_ends = [... % period_len ...]
```

Ultimately, this probably is just a matter of taste. Feel free to leave as-is.